Mathlib/GroupTheory/GroupAction/Basic.lean

```diff
@@ -106,6 +106,16 @@ theorem orbit.coe_smul {a : α} {m : M} {a' : orbit M a} : ↑(m • a') = m •
 #align mul_action.orbit.coe_smul MulAction.orbit.coe_smul
 #align add_action.orbit.coe_vadd AddAction.orbit.coe_vadd
 
+@[to_additive]
+lemma orbit_submonoid_subset (S : Submonoid M) (a : α) : orbit S a ⊆ orbit M a := by
```

Comment:
Can you generalise this lemma to `SubmonoidClass`?

Reply:
See the above resolved discussion. The `MulAction` instance doesn't exist for `SubmonoidClass`, and @tb65536's attempt to add it in #11304 ran into benchmark regressions.

Reply:
I do wonder whether a more limited version of #11304 might be viable.

Reply:
Can't you somehow *assume* the existence of the action?

Reply:
I suppose you could have another typeclass that asserts that a `SubmonoidClass` has a `MulAction` satisfying `smul_def`. I suspect it would be less convenient in various cases than having `smul_def` true by defeq, which is what defining the action for `SubmonoidClass` achieves.